File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    next_n = n + 1
    
    n_cnt = bin(n).count('1') 
    result_cnt = 0
    
    while(True):
        if n_cnt == bin(next_n).count('1'):
            return next_n
        
        next_n += 1 
        

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
import java.util.*;

class Solution {
    public int solution(int n) {
        
        String binaryN = Integer.toBinaryString(n);
        int nCnt = findOneCnt(binaryN);
        int nowNum = n + 1;
            
        while (true) {
            String binaryNum = Integer.toBinaryString(nowNum);
            
            if (nCnt == findOneCnt(binaryNum)) {
                int result = Integer.parseInt(binaryNum, 2);
                
                return result;
            }
            
            nowNum++;
        }
        
    }
    
    private int findOneCnt(String binaryNum) {
        int cnt = 0;
        
        for (int i = 0; i < binaryNum.length(); i++) {
            if (binaryNum.charAt(i) == '1') {
                cnt++;
            }
        }
        
        return cnt;
    }
    
    
}